src/openrc-run/openrc-run.c

```diff
@@ -448,9 +451,18 @@ svc_exec(const char *arg1, const char *arg2)
 				write_prefix(buffer, bytes, &prefixed);
 			}
 
-			/* Only SIGCHLD signals come down this pipe */
-			if (fd[0].revents & (POLLIN | POLLHUP))
+			/* Only SIGCHLD signals come down this pipe,
+			 * but with rc_parallel is might be another process */
+			if (fd[0].revents & (POLLIN | POLLHUP)) {
+				if (read(signal_pipe[0], &ret, sizeof(ret)) != sizeof(ret))
+					eerrorx("%s: receive exit code: %s",
+						service, strerror(errno));
+				ret = WEXITSTATUS(ret);
+				if (ret != 0 && errno == ECHILD)
+					/* killall5 -9 could cause this */
+					ret = 0;
```

Comment:
I think it makes sense to pull up the `service_pid = 0;` in here as well - though as far as I see it shouldn't make any difference (excluding any signal shenanigans).